### PR TITLE
Drag and drop spaces in the menu

### PR DESCRIPTION
**Pull request by valentinludu**
FIY: Because there is a loading time of 1 sec when I update the spaces order, I am disabling the spaces buttons in that second. I tested on a slow 3G with chrome and it takes 2 seconds. 
Thinking if a user is in Asia and has a slow connection, he will see the disabled buttons more then us.

### CODE REVIEW

**Comment by msadura** [comment]
> FIY: Because there is a loading time of 1 sec when I update the spaces order, I am disabling the spaces buttons in that second. I tested on a slow 3G with chrome and it takes 2 seconds. Thinking if a user is in Asia and has a slow connection, he will see the disabled buttons more then us.

hmm this feels hacky :D I mean why do we trigger loading of this much data jsut by reordering spaces? This feels fishy, and I would look for the cause of this and eliminate it. If you want I can help with debugging tomorrow.

**Review by Devorein — COMMENTED** [review]
Since we are using `setUser` after re-ordering user spaces, it causes a lot of requests to be sent (as they have [`user`] on their deps array). Probably we can store `spacesOrder` in a separate state to avoid updating the `user`? Otherwise we will have to make the `user` used inside the deps array to be much slimmer i.e just rely on `user.id`

**Review by msadura — APPROVED** [review]

**Comment by valentinludu** [comment]
> hmm this feels hacky :D I mean why do we trigger loading of this much data jsut by reordering spaces? This feels fishy, and I would look for the cause of this and eliminate it. If you want I can help with debugging tomorrow.

Had a discussion and we agreed to keep the buttons disabled until the API has finished loading.

File: lib/users/updateUserProfile.ts [hunk 1]
```diff
@@ -29,12 +27,13 @@ export async function updateUserProfile(userId: string, update: Partial<User>):
...
       data: {
-        avatar,
+        avatar: update.avatar,
         avatarChain: update.avatarChain,
         avatarContract: update.avatarContract,
         avatarTokenId: update.avatarTokenId,
         path: update.path,
-        email: update.email
+        email: update.email,
+        spacesOrder: update.spacesOrder
```
[the remark this answers is not in the document]
Comment (valentinludu) [reply lib/users/updateUserProfile.ts:36]: No, we don't. I will look into this to be sure we are not sending it.

[the remark this answers is not in the document]
Comment (valentinludu) [reply lib/users/updateUserProfile.ts:36]: I need just the update on the user table.

[the remark this answers is not in the document]
Comment (Devorein) [reply lib/users/updateUserProfile.ts:36]: Sorry I thought this was updating the user's mixpanel profile, my bad.

**Comment by mattcasey** [comment]
> Had a discussion and we agreed to keep the buttons disabled until the API has finished loading.
I'd like to hear the reasoning during standup

File: components/common/PageLayout/components/Sidebar/SidebarSubmenu.tsx [hunk 2]
```diff
@@ -67,7 +69,11 @@ export default function SidebarSubmenu({
   const currentSpace = useCurrentSpace();
   const { spaces, createNewSpace, isCreatingSpace } = useSpaces();
   const [spaceFormOpen, setSpaceFormOpen] = useState(false);
-  const { user } = useUser();
+  const { user, setUser } = useUser();
+  const { handleUserUpdate, isSaving } = useUserDetails({
```
Comment (mattcasey) [inline components/common/PageLayout/components/Sidebar/SidebarSubmenu.tsx:73]: useUserDetails is specifically for user profiles, where we need to load info for the profile page

**Review by mattcasey — APPROVED** [review]

File: components/common/PageLayout/components/Sidebar/SidebarSubmenu.tsx (hunk 2 above)
Comment (valentinludu) [reply components/common/PageLayout/components/Sidebar/SidebarSubmenu.tsx:73]: Keeping this since we don't need another hook that does the same thing.

**Pull request merged**